core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -36,15 +36,24 @@ sealed trait Cause[+E] extends Product with Serializable { self =>
       case Then(left, right) => left.died || right.died
       case Both(left, right) => left.died || right.died
       case Traced(cause, _)  => cause.died
+      case Meta(cause, _)    => cause.died
       case _                 => false
     }
 
+  /**
+   * Returns the `Throwable` associated with the first `Die` in this `Cause` if
+   * one exists.
+   */
+  final def dieOption: Option[Throwable] =
+    fold(_ => None, t => Some(t), None)(_ orElse _, _ orElse _, (z, _) => z)
```

Comment:
Might want to _always_ use named parameters with the new fold 😆

Reply:
Good idea. The ordering of type parameters is definitely arbitrary other than that I tried to curry it so that the `Z` parameter could be inferred.